scripts/ipfs/ipfs-setup.sh

```diff
@@ -0,0 +1,100 @@
+    sleep 1
+    attempts=$((attempts-1))
+done
+
+if [[  ${attempts} -eq 0 ]]; then
+    echo "Error: Failed to connect to local IPFS daemon. Is it running?"
+    exit 1
+fi
+
+# Update peers's data since IPFS just started
+echo "Updating nodeinfo..."
+sudo /usr/local/bin/nodeinfo-update.sh
+
+function addPeer  {
+    addr=$1
+    # See if they have IPFS enabled
+    res=$(curl http://["${addr}"]/nodeinfo.json -s)
+    if [ ! -x "${res}" ]; then
+        id=$(echo "${res}" | jq -r -M '.services.ipfs.ID')
+        # Value is found
+        if [[ ! ${id} == "null" ]] && [[ ! "${id}" == "" ]]; then
+            # Connect to neighbouring IPFS nodes
+            # Check for QUIC connections first
+            if [ "$(echo "${res}" | jq -r -M '.services.IPFS.quic_enabled')" == 'true' ]; then
+                # ID is not needed for QUIC connections
+                ipfs swarm connect "/ip6/${addr}/udp/4001/quic"
+                echo "Connecting to ${addr} with QUIC"
+            else
+                ipfs swarm connect "/ip6/${addr}/tcp/4001/ipfs/${id}"
+                echo "Connecting to ${addr} regularly"
+            fi
+        fi
+    fi
+}
+
+# Continually mesh with peers and add or remove filter rules as networks go up and down
+while true; do
+    # Check if CJDNS is running
+    if [ "$(systemctl status cjdns.service | grep 'Active: ' | awk '{ print $2 }')" = 'active' ]; then
+        # Connect with direct peers over IPFS
+        while read -r cjdns_peer; do
+            cjdns_addr=$(sudo /opt/cjdns/publictoip6 "$cjdns_peer")
+            addPeer "${cjdns_addr}"
+
+            # Add all that node's peers to the bottom of the list to check further hop peers
+            # XXX: The below command hasn't been working -- so for now only 1-hop peers are checked
+            #peers+=$(cjdnstool query getpeers $peer | sed -e '1d;$d' |awk -F. '{ print $6".k" }')
+
+        done <<< "$(sudo nodejs /opt/cjdns/tools/peerStats 2>/dev/null | awk '{ if ($3 == "ESTABLISHED") print $2 }' | awk -F. '{ print $6".k" }' | xargs)"
+
+        # Remove a filter if there was one
+        ipfs swarm filters rm '/ip6/fc00::/ipcidr/8'
+        echo "Enabled access to CJDNS network"
+    else
+        ipfs swarm filters add '/ip6/fc00::/ipcidr/8'
+        echo "Blocked CJDNS network"
+    fi
+
+    if [ "$(systemctl status yggdrasil.service | grep 'Active: ' | awk '{ print $2 }')" = 'active' ]; then
+        while read -r ygg_peer; do
+            addPeer "${ygg_peer}"
+        done <<< "$(sudo yggdrasilctl getPeers | grep -v "(self)" | awk '{print $1}' | grep -v bytes_recvd | xargs)"
+
+        ipfs swarm filters rm '/ip6/200::/ipcidr/7'
+        echo "Enabled access to Yggdrasil network"
+    else
+        ipfs swarm filters add '/ip6/200::/ipcidr/7'
+        echo "Blocked Yggdrasil network"
+    fi
+
+    # Clearnet or regular Internet IPv4 access
+    if ! ping -w 2 1.1.1.1 &>/dev/null; then
```

Comment:
This type of Internet check is usually problematic. What is the motivation behind adding this filter at all?